File: AtlasWeb/Core/Modules/Tibbi_Surec/PatientSearch/PatientSearchFormViewModel.ts

import { Guid } from "NebulaClient/Mscorlib/Guid";
import { Patient } from 'NebulaClient/Model/AtlasClientModel';
import { Appointment } from 'NebulaClient/Model/AtlasClientModel';
import { Type } from 'NebulaClient/ClassTransformer';
import { BaseViewModel } from 'NebulaClient/Model/BaseViewModel';

export class PatientSearchFormViewModel extends BaseViewModel {

    public SearchText: string = "";
    @Type(() => BasePatientInfoListsModel)
    public PatientInfoLists: BasePatientInfoListsModel;
    @Type(() => BasePatientSearchResultModel)
    public PatientSearchResult: BasePatientSearchResultModel;
    @Type(() => Guid)
    public SelectedPatientObjectID: Guid;
    @Type(() => BasePatientInfoDet)
    public SelectedPatient: BasePatientInfoDet;


    constructor() {
        super();
        this.PatientSearchResult = new BasePatientSearchResultModel();
        this.PatientInfoLists = new BasePatientInfoListsModel();
        this.SelectedPatient = new BasePatientInfoDet();

    }
}

export class BasePatientSearchResultModel {


    @Type(() => Patient)
    public PatientInfo: Patient;
    @Type(() => Appointment)
    public AppointmentInfo: Appointment;
   // public PatientAdmissionInfo: PatientAdmission;

    public PatientIDandFullName: string;
    public PatientRefNo: string;
    public PatientPhoneNumber: string;

    constructor() {
    }
}

export class BasePatientInfoListsModel {

    @Type(() => BasePatientInfoDet)
    public PatientList: BasePatientInfoDet[];
   // public PatientAdmissionList: BasePatientAdmissionInfoDet[];
     //public PatientInfo: Patient;
     //public AppointmentInfo: Appointment;
     //public AdmissionSpecOrResOfCurrentDayEpisodes: string = "";
     //public PatientAdmissionInfo: PatientAdmission;

    constructor() {
        this.PatientList = new Array<BasePatientInfoDet>();
       // this.PatientAdmissionList = new Array<BasePatientAdmissionInfoDet>();
    }
}

 export class BasePatientInfoDet
 {
     @Type(() => Guid)
    public ObjectID: Guid ;
    public NameSurname: string;
    public Info: string;

}
// export class PatientAdmissionInfoDet
//{
//     public ObjectID: Guid;
//     public ActionDate: string;
//     public Info: string;
//}
// class AppointmentInfo
//{
//    public  ObjectID: Guid;
//}

File: AtlasWeb/Core/Modules/Tibbi_Surec/PatientSearch/PatientSearchFormViewModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using TTObjectClasses;

namespace Core.Models
{
    public class PatientSearchFormViewModel
    {
        public string SearchText
        {
            get;
            set;
        }

        public BasePatientSearchResultModel PatientSearchResult
        {
            get;
            set;
        }

        public BasePatientInfoListsModel PatientInfoLists
        {
            get;
            set;
        }

        public PatientInfoDet SelectedPatient
        {
            get;
            set;
        }

        public PatientSearchFormViewModel()
        {
            this.PatientSearchResult = new BasePatientSearchResultModel();
            this.PatientInfoLists = new BasePatientInfoListsModel();
            //this.SelectedAdmission = new PatientAdmissionInfoDet();
            this.SelectedPatient = new PatientInfoDet();
        }
    }

    public class BasePatientInfoListsModel
    {
        public List<PatientInfoDet> PatientList
        {
            get;
            set;
        }

        //public Patient PatientInfo { get; set; }
        //public Appointment AppointmentInfo { get; set; }
        // public PatientAdmission PatientAdmissionInfo { get; set; }
        public BasePatientInfoListsModel()
        {
            this.PatientList = new List<PatientInfoDet>();
        //this.PatientInfo = new Patient();
        //this.AppointmentInfo = new Appointment();
        //this.PatientAdmissionInfo = new PatientAdmission();
        }
    }

    public class BasePatientSearchResultModel
    {
        public Patient PatientInfo
        {
            get;
            set;
        }

        public Appointment AppointmentInfo
        {
            get;
            set;
        }

        public string PatientIDandFullName
        {
            get;
            set;
        }

        public string PatientRefNo
        {
            get;
            set;
        }

        public string PatientPhoneNumber
        {
            get;
            set;
        }
    }

    public class PatientInfoDet
    {
        public Guid ObjectID
        {
            get;
            set;
        }

        public string NameSurname
        {
            get;
            set;
        }

        public string Info
        {
            get;
            set;
        }

        public string FullInfo
        {
            get;
            set;
        }

        public string FatherName
        {
            get;
            set;
        }

        public string Dob
        {
            get;
            set;
        }

        public string CityOfBirth
        {
            get;
            set;
        }

        public bool? Privacy
        {
            get;
            set;
        }
    }
//public class AppointmentInfo
//{
//    public Guid ObjectID { get; set; }
//}
}